### PR TITLE
ISSUE-905 Test Mode: Need API modifications to delete all test source…

### LINKED ISSUE #905: Test Mode: Need API modifications to delete all test source/sink when test case is deleted

### PR DESCRIPTION
**Pull request by HeartSaVioR**
…/sink when test case is deleted

* remove corresponding sources and sinks when removing test case

This closes #905 

@shahsank3t 
Please apply the patch and try to delete test case. 
Like other entities, the API endpoint path for deleting test case is same as getting specific test case. You just need to change the method to DELETE. (`/topologies/{topologyId}/testcases/{testCaseId}`)

@arunmahadevan @shahsank3t 
Please review and comment. Thanks in advance!

### CODE REVIEW

File: streams/service/src/main/java/com/hortonworks/streamline/streams/service/TopologyTestRunResource.java [hunk 1]
```diff
@@ -509,7 +520,7 @@ public Response getTestRunCaseSinkByTopologySink(@PathParam("topologyId") Long t
     @Path("/topologies/{topologyId}/testcases/{testCaseId}/sinks")
     public Response listTestRunCaseSink(@PathParam("topologyId") Long topologyId,
                                           @PathParam("testCaseId") Long testCaseId) {
-        Collection<TopologyTestRunCaseSink> sources = catalogService.listTopologyTestRunCaseSink(topologyId, testCaseId);
+        Collection<TopologyTestRunCaseSink> sources = catalogService.listTopologyTestRunCaseSink(testCaseId);
```
Comment (arunmahadevan) [inline streams/service/src/main/java/com/hortonworks/streamline/streams/service/TopologyTestRunResource.java]: note sources -> sink

**Comment by arunmahadevan** [comment]
+1

File: streams/service/src/main/java/com/hortonworks/streamline/streams/service/TopologyTestRunResource.java (hunk 1 above)
Comment (HeartSaVioR) [reply streams/service/src/main/java/com/hortonworks/streamline/streams/service/TopologyTestRunResource.java]: Nice finding. Fixed.

**Comment by HeartSaVioR** [comment]
@arunmahadevan Squashed the commits.

**Pull request merged**